internal/util/debug/debug.go

```diff
@@ -123,3 +142,80 @@ func RunHandler(ctx context.Context, addr string, r prometheus.Registerer, l *za
 	s.Close()
 	l.Sugar().Info("Debug server stopped.")
 }
+
+func getArchiveHandler() http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		var (
+			scheme          string
+			u               url.URL
+			responses       = make(map[string][]byte, 2)
+			debugFilePrefix = "FerretDB"
+			debugFileName   = "debug.zip"
+			err             error
+		)
+
+		u.Path = metricsPath
+		u.Host = req.Host
+
+		if req.URL.Scheme == "" {
+			scheme = "http"
+		}
+
+		u.Scheme = scheme
+
+		metricsFileName := generateFileName(debugFilePrefix, filepath.Base(metricsPath)+".txt")
+		responses[metricsFileName], err = performRequest(u)
+		if err != nil {
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		u.Path = pprofPath
+		pprofFileName := generateFileName(debugFilePrefix, filepath.Base(pprofPath)+".html")
+		responses[pprofFileName], err = performRequest(u)
```

Comment:
We should get a memory profile, not HTML page with a list of profiles

Reply:
I'm not sure why we decided to parse HTML, but we definitely should not do that. The URL for the memory profile is fixed: `debug/pprof/heap` (without query parameters).

Also, please don't resolve conversations: https://github.com/FerretDB/FerretDB/blob/c8f8c6851e4a07ae5cc9297ae2a5019bcea6c7f1/CONTRIBUTING.md?plain=1#L358-L362

It might help to re-read the whole file.

Reply:
Hi @AlekSi  ,HTML parsing is required since we add dynamic resource tracking in pprof, I am assuming we need them as well in zip. if not , we should improve the task by specifying what's exactly required.  Additionally Debug=1 , exports the data in the legacy text format. I tried exporting few files without this option and they were binary , some non readable info.

Reply:
Do we only one URL - "debug/pprof/heap" to be grabbed and nothing else.

Reply:
> we add dynamic resource tracking in pprof

Why do we need that?

> Do we only one URL - "debug/pprof/heap" to be grabbed and nothing else.

Yes.

> Additionally Debug=1 , exports the data in the legacy text format. I tried exporting few files without this option and they were binary , some non readable info.

That format is readable with [`go tool pprof`](https://go.dev/doc/diagnostics#profiling) and contains more information than the legacy text format. We need that.

Reply:
Sure. Updated the code to gather only two files in zip. matrics and heap. test updated accordingly.

Reply:
@noisersup PTAL ^ We don't need `debug=1`